app/grandchallenge/core/utils/grand_challenge_forge.py

```diff
@@ -21,35 +35,45 @@ def get_forge_json_description(challenge, phase_pks=None):
     def process_archive(archive):
         return {
             "slug": archive.slug,
-            "url": reverse("archives:detail", kwargs={"slug": archive.slug}),
-        }
-
-    def process_component_interface(component_interface):
-        return {
-            "slug": component_interface.slug,
-            "kind": component_interface.get_kind_display(),
-            "super_kind": component_interface.super_kind.label,
-            "relative_path": component_interface.relative_path,
+            "url": archive.get_absolute_url(),
         }
 
     def process_phase(phase):
         return {
             "slug": phase.slug,
             "archive": process_archive(phase.archive),
             "algorithm_inputs": [
-                process_component_interface(ci)
+                _process_component_interface(ci)
                 for ci in phase.algorithm_inputs.all()
             ],
             "algorithm_outputs": [
-                process_component_interface(ci)
+                _process_component_interface(ci)
                 for ci in phase.algorithm_outputs.all()
             ],
         }
 
     return {
         "challenge": {
             "slug": challenge.slug,
+            "url": challenge.get_absolute_url(),
             "phases": [process_phase(p) for p in phases],
             "archives": [process_archive(a) for a in archives],
         }
     }
+
+
+def get_forge_algorithm_template_context(algorithm):
+    return {
+        "algorithm": {
+            "title": algorithm.title,
+            "url": algorithm.get_absolute_url(),
+            "inputs": [
+                _process_component_interface(ci)
+                for ci in algorithm.inputs.all()
+            ],
+            "outputs": [
+                _process_component_interface(ci)
+                for ci in algorithm.outputs.all()
+            ],
+        }
+    }
```

Comment:
Don't you also want the `slug` in here?

Reply:
What do you imagine we'd need the `slug` for? We won't actually be interfacing with the algorithm on GC. 

I had the idea in my head we might re-slug the title since we are working locally anyway. That way the slug keeps pointing to the algorithm. There is the trouble with duplicates then, but I honestly think the chances of that are so tiny that it's acceptable to keep things recognizable locally.

Reply:
I didn't have anything in particular in mind, I just figured that since the slug is the unique, human readable identifier of the algorithm it probably should be in there (just like you do for the phases and archives), but I trust your intuition here if you think you don't need it.

Reply:
But just for my understanding: why re-slug the title if you could just pass along the slug? What do you need the title for?

Reply:
Slug is only created at creation time and can never be updated. The title is the most current 'human' name that the algorithm carries. We see that a lot in challenge phases, I think it also applies to algorithms.